src/fmu/dataio/dataio.py

```diff
@@ -886,17 +899,23 @@ def export(self, obj, return_symlink=False, **kwargs) -> str:
         outfile = Path(metadata["file"]["absolute_path"])
         metafile = outfile.parent / ("." + str(outfile.name) + ".yml")
 
-        useflag: Union[bool, str]
-        if isinstance(obj, pd.DataFrame):
-            useflag = self.table_include_index
-        else:
-            useflag = self._usefmtflag
+        useflag = (
+            self.table_include_index
+            if isinstance(obj, pd.DataFrame)
+            else self._usefmtflag
+        )
 
         obj = self._check_obj_if_file(obj)
         logger.info("Export to file and compute MD5 sum, using flag: <%s>", useflag)
-        outfile, md5 = export_file_compute_checksum_md5(
-            obj, outfile, outfile.suffix, flag=useflag
+        toutfile, md5 = export_file_compute_checksum_md5(
+            obj,
+            outfile,
+            outfile.suffix,
+            flag=useflag,  # type: ignore
+            # BUG(?): Looks buggy, if flag is bool export_file will blow up.
```

Comment:
Might be worth making an issue over this checksum function, perhaps a few things that could be improved based upon your comments in here

Reply:
Yeah, i have a few other potential bugs that need tickets as well.

Reply:
https://github.com/equinor/fmu-dataio/issues/410